content/en/docs/ibm/deploy/authentication.md

```diff
@@ -110,38 +110,165 @@ column):
+
+1. First, update the Knative domain that is used for the KFServing routes to the
+hostname that you used when updating `kubeflow-gateway`.
+
+    ```shell
+    kubectl edit configmap config-domain -n knative-serving
+    ```
+
+    This will open your default text editor, and you will see something like:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      _example: |
+        ################################
+        #                              #
+        #    EXAMPLE CONFIGURATION     #
+        #                              #
+        ################################
+        # ...
+        example.com: |
+    kind: ConfigMap
+    ...
+    ```
+
+    Add a line above the `_example` key with your hostname as the key and an empty string value.
+    Be sure to update `<hostname>`:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      <hostname>: ""
+      _example: |
+       ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Then, save and exit. The routes for your InferenceServices will start using this new domain.
+
+2. Since the certificates provided by IBM Cloud only allow for a single level of domain name added
+to the base domain, the domain template for Knative needs to be adjusted so that the certificates
+will be valid for the InferenceService routes.
+
+    ```shell
+    kubectl edit configmap config-network -n knative-serving
+    ```
+
+    This will open your default text editor, and you will again see something like:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      _example: |
+        ################################
+        #                              #
+        #    EXAMPLE CONFIGURATION     #
+        #                              #
+        ################################
+        # ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Add a line above the `_example` key with the `domainTemplate` key like the following:
+
+    ```YAML
+    apiVersion: v1
+    data:
+      domainTemplate: "{{.Name}}-{{.Namespace}}.{{.Domain}}"
+      _example: |
+       ...
+    kind: ConfigMap
+    ...
+    ```
+
+    Save and exit. The default template uses a two-level subdomain (i.e. `{{.Name}}.{{.Namespace}}.{{.Domain}}`),
+    so this just adjusts it to use one.
+
+
+3. Adjust `kubeflow-gateway` one more time, adding a wildcard host in the HTTPS `hosts` section.
+
+    ```shell
+    kubectl edit gateway kubeflow-gateway -n kubeflow
+    ```
+
+    This will open your default text editor, but you can also optionally edit `kubeflow-gateway.yaml` file
+    you created previously. Here, just add another entry to the HTTPS `hosts` list containing your hostname prepended
+    with a `*.` so that the Knative subdomains are properly routed.
+
+    ```YAML
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: kubeflow-gateway
+      namespace: kubeflow
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+      - hosts:
+        - '<hostname>'
+        - '*.<hostname>'
```

Comment:
Have you tried the sudomain works with a deployed Kubeflow with authentication? I assume AppID will redirect the request to the configured FQDN without the subdomain.

Reply:
Hi @shawnzhu, thanks for the review. I can certainly split this PR up to separate the scopes.

I have tested, and something like the following curl command works:

```
curl -v https://sklearn-iris-bar-p-test.kfp-dev-442dbba0442be6c8c50f31ed96b00601-0001.sjc03.containers.appdomain.cloud/v1/models/sklearn-iris-bar:predict -d '{"instances": [[6.8,  2.8,  4.8,  1.4],[6.0,  3.4,  4.5,  1.6]]}' -H "Authorization: Bearer eyJhb..."

{"predictions": [1, 1]}
```

Without the Authorization header, I get a HTTP 302, and then what I am assuming is a redirect to the auth UI.

KFserving does not have it's own gateway, it either uses Knative's gateway (`knative-ingress-gateway`) if standalone or `kubeflow-gateway` if KFServing is installed with the rest of Kubeflow via the manifests.

Reply:
> Without the Authorization header, I get a HTTP 302, and then what I am assuming is a redirect to the auth UI

that's what I mentioned above. it would (as designed) pass through the envoyfilter listens at gateway level when providing an authorization header. (which authservice image did you use? it would be very helpful to share which kfdef you've used to deploy this kubeflow).

Reply:
The KFDef used was this one: https://raw.githubusercontent.com/kubeflow/manifests/v1.2-branch/kfdef/kfctl_ibm_multi_user.v1.2.0.yaml 

I did get Kfserving predict requests working just using HTTP and without the need for any auth headers (thanks to @yhwang) by adding the following to `kubeflow-gateway`.

```
- hosts:
  - '*.<hostname>'
  port:
      name: http-kfserving
      number: 80
      protocol: HTTP
```

Then updating the envoyfilter (authn-filter) to just listen on port 443:
```
    listenerMatch:
      portNumber: 443
      listenerType: GATEWAY 
```
However, I am unsure if these adjustments should be included here.

Reply:
@pvaneck Thanks! that works!